### PR TITLE
Membersforquery fixes

### PR DESCRIPTION
**Pull request by Dieterbe**
TestPeersForQueryMulti occasionally fails in CI. I've seen this in various non-cluster-related PR's and I think also in master.  I think this has been more frequent recently, but i'm not sure.
when I run `for i in $(seq 1 200); do go test -run=TestPeersForQueryMulti >> master.txt; done` I observed 2 failures, 198 passes.

see the indiviual commit messages for how i've gone about addressing this.

### CODE REVIEW

File: cluster/cluster.go [hunk 1]
```diff
@@ -142,7 +145,8 @@ LOOP:
 		}
 		// if no nodes have been selected yet then grab a
 		// random node from the set of available nodes
-		selected := candidates.nodes[rand.Intn(len(candidates.nodes))]
+		pos := int(atomic.AddUint64(&counter, 1)) % len(candidates.nodes)
```
Comment (replay) [inline cluster/cluster.go]: should that counter maybe get reset to `0` at some point? I know `Uint64` is big, but being optimistic we should assume MT can have a long life time without restarting :)

Comment (Dieterbe) [reply cluster/cluster.go]: when it wraps around, all the logic keeps on functioning fine as far as i can see.
in fact i might as well have used a uint32 for that reason.

Comment (replay) [reply cluster/cluster.go]: yeah true

**Comment by Dieterbe** [comment]
i just realized the suggested counter-remainder approach may result in a skewed distribution , need to fine tune it a bit more

File: cluster/cluster.go [hunk 2]
```diff
@@ -140,9 +145,10 @@ LOOP:
 				continue LOOP
 			}
 		}
-		// if no nodes have been selected yet then grab a
-		// random node from the set of available nodes
-		selected := candidates.nodes[rand.Intn(len(candidates.nodes))]
+		// if no nodes have been selected yet then grab a node from
+		// the set of available nodes in such a way that nodes are
+		// weighted fairly across MembersForQuery calls
+		selected := candidates.nodes[count%len(candidates.nodes)]
```
Comment (replay) [inline cluster/cluster.go]: is reading the count like this atomic?

Comment (replay) [reply cluster/cluster.go]: ah, sorry, got it now. the `count` is the return value of the above atomic operation

**Pull request merged**